Exercises/Tom_Knoden/Woche08_Aufgaben_Version2.py

```diff
@@ -0,0 +1,72 @@
+# Prepare an analysis of the titanic dataset (data/titanic.csv).
+# You are interested in determining factors that lead to death or survival of passengers.
+# However, before actually training an ML model to predict death or survival, you should make
+# yourself familiar with the data.
+#
+# Look into which features are available, what types of features these are, whether
+# values are missing and how to deal with them, etc.
+# You can then continue to create some visualizations of things you think might be interesting or relevant
+# in predicting survival, or just generally useful to know about the dataset.
+# The output should not just be prints of numbers or figures, but also some explanatory text of
+# what you have been analysing, and which conclusions you can draw from the different steps of your analysis.
+#
+# You can do all of this in a Jupyter Notebook, embedding analysis, code and explanations in a single document.
+# Alternatively you can create a separate script for the analysis, and then list results, figures and explanations
+# either in the PR directly, or in some form of pdf/word/markdown/... document.
+#
+# For this exercise I very much encourage you to work together to come up with ideas of which
+# calculations/plots/... might be interesting to analyse. You will find very many such analyses online
+# for the titanic data, but I would recommend trying to think of your own ideas first, and only later
+# look for more inspiration online.
+
+import pandas as pd
+import matplotlib.pyplot as plt
+from pathlib import Path
+
+input_file_path = Path('data/titanic.csv')
+titanic_data = pd.read_csv(input_file_path)
```

Comment:
this way of reading in the file is not working for me. you could use something like
```suggestion
root_dir = pathlib.Path(__file__).absolute().parents[2]
with open(root_dir / "data" / "titanic.csv", "r") as ifile:
    titanic_data = pd.read_csv(ifile)
```
instead.